File: python/problem_5.py
args = (1, 21)

def gcd(a, b):
    while b:      
        a, b = b, a % b
    return a

def lcm(a, b):
    return a * b // gcd(a, b)

def lcmm(*a):
    return reduce(lcm, a)

def problem_5(lower, upper):
    return lcmm(*xrange(lower, upper))

File: Scala/problem5.scala
import scala.math.BigInt

object Main extends App {
  
  def gcd[T](a: T, b: T): T = {
    if(b == 0) a
    else gcd(b, a % b)
  }
  
  def lcm[T](a: T, b: T): T = {
    a * b / gcd(a, b)
  }
  
  def lcmm[T](xs: List[T]): T = {
    xs reduceLeft lcm
  }
  
  def problem5[T](rng: Range) = lcmm[T](rng.toList)
  
}
  
